packages/block-editor/src/components/block-edit/edit.js

```diff
@@ -48,9 +50,14 @@ const Edit = ( props ) => {
 const EditWithFilters = withFilters( 'editor.BlockEdit' )( Edit );
 
 const EditWithGeneratedProps = ( props ) => {
-	const { attributes = {}, name } = props;
+	const { clientId, attributes = {}, name } = props;
 	const blockType = getBlockType( name );
 	const blockContext = useContext( BlockContext );
+	// Sync the block context with the block editor store.
+	useDispatch( blockEditorStore ).updateBlockContext(
+		clientId,
+		blockContext
+	);
```

Comment:
We can't call actions on render, must be used to an effect, see https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js

Reply:
Also, we shouldn't do this for every block, we should only do if where context it provided

Reply:
Actually, looking at packages/block-editor/src/components/inner-blocks/use-block-context.js, it appears block context is just a selector? So we could derive the whole thing the getBlockAttributes selector?

Reply:
So in `getBlockAttributes`, we should loop up the parents, see if they provide context, and if so set their attributes as context?

Reply:
Good points 🙂 

I thought we needed to get the value from `useContext( BlockContext )`, as you were doing [here](https://github.com/WordPress/gutenberg/pull/60826/files#diff-99d79bdba99fe4517413b0fd9503eef924e439c03d44504c1ad0606b727cebb6R62), but maybe you are right and we can do that looping up the parents. I can give that a try.

Reply:
That wouldn't include context provided by `BlockContextProvider` that doesn't belong to a block, right?

I'm thinking that, if I am not mistaken, we are providing `postId` and `postType` as context in the post template: [link](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/post-template/edit.js#L212-L220). But that wouldn't be available iterating through the parents, right?

Reply:
Hm, I guess then it is easier to sync the context. I'd sync it inside of `BlockContextProvider`. Then create a selector that gets the context which would loop through the parents and finds/merges the context.

Reply:
Do you mean something like [this try](https://github.com/WordPress/gutenberg/pull/61895/commits/c93906123cedcd6aaed84d443b3eb0f60af308cb)?